### PR TITLE
Added tests for src/resolver/mutation/addUserToUserFamily.ts

### LINKED ISSUE #1817: Tests: src/resolvers/Mutation/addUserToUserFamily.ts
The Talawa-API code base needs to be 100% reliable. This means we need to have 100% test code coverage.

1. Tests need to be written for file `src/resolvers/Mutation/addUserToUserFamily.ts`
1. We will need the API to be refactored for all methods, classes and/or functions found in this file for testing to be correctly executed.
1. When complete, all all methods, classes and/or functions in the refactored file will need to be tested. 
1. These tests must be placed in a single file with the same path as the tested file, but under the `tests/` with a `.spec.ts` suffix.
1. You may need to create the appropriate directory structure to do this.

### IMPORTANT:

Please refer to the parent issue on how to implement these tests correctly:

- https://github.com/PalisadoesFoundation/talawa-api/issues/490

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base. This will be clearly visible when you submit your PR.
- [The current code coverage for the file can be found by visting this page](https://app.codecov.io/gh/PalisadoesFoundation/talawa-api?displayType=list). Login using your GitHub credentials.
- Create your own `codecov.io` to help with testing.
- The PR you create will show a report for the code coverage for the file you have added. You can also use that as a guide.

### Other

Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue:

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by NayOoLwin5**
**What kind of change does this PR introduce?**

It increases `addUserToUserFamily` resolver to 100%

**Issue Number:**

Fixes #1817 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

codecov1
codecov2

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

File: src/resolvers/Mutation/addUserToUserFamily.ts [hunk 1]
```diff
@@ -52,9 +52,9 @@ export const addUserToUserFamily: MutationResolvers["addUserToUserFamily"] =
     //check whether user is admin of the family
     await adminCheck(currentUser?._id, userFamily);
 
-    const isUserMemberOfUserFamily = userFamily.users.some((user) => {
-      user.equals(args.userId);
-    });
+    const isUserMemberOfUserFamily = !!userFamily.users.find((user) =>
+      user.equals(args.userId),
+    );
```
Comment (EshaanAgg) [inline src/resolvers/Mutation/addUserToUserFamily.ts:57]: Why this change? Aren't they equivalent? The earlier one has better readability IMO.

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
PTAL

**Comment by palisadoes** [comment]
@NayOoLwin5 Please respond to the reviewer comments

File: src/resolvers/Mutation/addUserToUserFamily.ts (hunk 1 above)
Comment (NayOoLwin5) [reply src/resolvers/Mutation/addUserToUserFamily.ts:57]: That some method always returns false even if two objectIds are same which is the reason the validation code wasn't tested at all (not covered). @EshaanAgg

**Comment by palisadoes** [comment]
We would like to merge this PR, however you haven’t addressed the reviewer questions. Please address each one and mark them as resolved when the work is completed. Only then will they be able to approve your work so that it can be merged into the code base.

**Comment by NayOoLwin5** [comment]
@palisadoes @EshaanAgg I have resolved the conversation. Please take a look.
Sorry for the late response.

**Comment by palisadoes** [comment]
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by palisadoes** [comment]
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by Cioppolo14** [comment]
@NayOoLwin5 Please fix the conflicting file.

**Comment by palisadoes** [comment]
Closing as abandoned
